syntax = "proto3";

package observatory.protobuf;

message RecordMetricRequest {
    string model = 1;
    int32 version = 2;
    string experiment = 3;
    string run_id = 4;
    string metric = 5;
    int64 timestamp = 6;
    float value = 7;
}

message RecordMetricResponse {
    int32 status = 1;
}

message RecordSessionStartRequest {
    string model = 1;
    int32 version = 2;
    string experiment = 3;
    string run_id = 4;
    int64 timestamp = 5;
}

message RecordSessionStartResponse {
    int32 status = 1;
}

message RecordSessionCompletionRequest {
    string model = 1;
    int32 version = 2;
    string experiment = 3;
    string run_id = 4;
    int64 timestamp = 5;
    string status = 6;
}

message RecordSessionCompletionResponse {
    int32 status = 1;
}

message RecordSettingsRequest {
    string model = 1;
    int32 version = 2;
    string experiment = 3;
    string run_id = 4;
    string data = 5;
}

message RecordSettingsResponse {
    int32 status = 1;
}

message Chunk {
    string model = 1;
    int32 version = 2;
    string experiment = 3;
    string run_id = 4;
    string filename = 5;
    bytes buffer = 6;
}

message RecordOutputResponse {
    int32 status = 1;
}

service TrackingService {
    rpc RecordMetric(RecordMetricRequest) returns (RecordMetricResponse);
    rpc RecordSessionStart(RecordSessionStartRequest) returns (RecordSessionStartResponse);
    rpc RecordSessionCompletion(RecordSessionCompletionRequest) returns (RecordSessionCompletionResponse);
    rpc RecordSettings(RecordSettingsRequest) returns (RecordSettingsResponse);
    rpc RecordOutput(stream Chunk) returns (RecordOutputResponse);
}